### PR TITLE
Include FTMS bike types

### PR DESCRIPTION
**Pull request by bryanoostra**

### CODE REVIEW

File: sdk-ftms/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# `com.nohrd.bike:sdk-ftms`
+
+The `com.nohrd.bike:sdk-ftms` artifact contains the sources for
+reading data from a connected FTMS (FiTness Machine Service) Bike.
+
+## Usage
```
Comment (bryanoostra) [inline sdk-ftms/README.md:6]: include this

File: dependencies.gradle [hunk 2]
```diff
@@ -1,11 +1,11 @@
-ext.composeVersion = "1.0.0-alpha07"
+ext.composeVersion = "1.0.0-alpha10"
```
Comment (nhaarman) [inline dependencies.gradle:1]: Use the most recent version

File: build.gradle [hunk 3]
```diff
@@ -6,7 +6,7 @@ buildscript {
         maven { url "https://plugins.gradle.org/m2/" }
     }
     dependencies {
-        classpath "com.android.tools.build:gradle:4.2.0-alpha16"
+        classpath 'com.android.tools.build:gradle:7.0.0-alpha04'
```
Comment (nhaarman) [inline build.gradle]: Use `"` instead of `'` and use `7.0.1`

**Review by nhaarman — CHANGES_REQUESTED** [review]

File: dependencies.gradle (hunk 2 above)
Comment (bryanoostra) [reply dependencies.gradle:1]: Tried that yesterday, it triggered more updates and in the end functionality disappeared, so I reverted it and called it a day.

File: build.gradle (hunk 3 above)
Comment (bryanoostra) [reply build.gradle]: 👍🏽

**Review by nhaarman — CHANGES_REQUESTED** [review]
Some remarks about the structure. 

The `sdk` module is meant to serve the original v1 version of the NOHrD Bike and perform its calculations. The `sdk-ble` module provides the BLE Services and Characteristics required for the bluetooth connection with that Bike. Just like that, you could imagine an `sdk-usb` version for the cabled connection with that Bike.

Now we're introducing a new Bike, a NOHrD Bike v2 you could say, which includes the FTMS service. The implementation for this Bike should stand completely separate from the v1 version.

In this PR you've included the FTMS artifact as a dependency of the BLE artifact, mixing v1 and v2:
image

A better approach would be to keep the `sdk-ftms` and `sdk-ble` separated, to communicate their different intents better. This means that users of this library must depend on two separate artifacts:

image

I think we should then also think about the naming of the artifacts to communicate their difference. `com.nohrd.bike:sdk` could become `com.nohrd.bike:sdk-v1` or `com.nohrd.bike:sdk-original` (or some other defining name for the original NOHrD Bike); `com.nohrd.bike:sdk-ble` becomes `com.nohrd.bike:sdk-v1-ble` or `com.nohrd.bike:sdk-original-ble`; `com.nohrd.bike:sdk-ftms` can then stay the same.

We should also release this as version `2.0`.

File: domain/src/main/kotlin/com/nohrd/bike/domain/BikeDataListener.kt [hunk 4]
```diff
@@ -1,4 +1,4 @@
-package com.nohrd.bike.sdk
+package com.nohrd.bike.domain
```
Comment (nhaarman) [inline domain/src/main/kotlin/com/nohrd/bike/domain/BikeDataListener.kt:1]: Don't share this between the `sdk` and `sdk-ftms` modules. The implementation for these two modules have very different reasons to change, even though the listeners behave the same right now. It is the client's responsibility to adapt these interfaces to their needs.

**Pull request closed without merging**